airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/integrations/destination/async/deser/AirbyteMessageDeserializer.kt

```diff
@@ -53,7 +53,7 @@ class AirbyteMessageDeserializer(
             partial.record?.data = null
         } else if (AirbyteMessage.Type.STATE == msgType) {
             partial.withSerialized(message)
-        } else {
+        } else if (AirbyteMessage.Type.TRACE != msgType) {
             throw RuntimeException(String.format("Unsupported message type: %s", msgType))
```

Comment:
Should we start not failing for unknown msg types, starting from this CDK version ?

Reply:
ooh, that's an interesting idea. lemme see how hard this would be (... mostly just not sure how easy it is to verify that I didn't miss a spot)

Reply:
blegh. we actually need to fix how we deserialize messages. I can remove this throw, but jackson fails to deser a message with unrecognized enum value

which gets us _partway_ there, but not entirely. I'm going to leave https://github.com/airbytehq/airbyte-internal-issues/issues/8064 open and add some notes in there.

Reply:
also, added a test for AsyncStreamConsumer to at least enforce that we don't throw on `log` messages :P

Reply:
👍